coffee_github/src/repository.rs

```diff
@@ -310,6 +310,27 @@ impl Repository for Github {
         }
     }
 
+    async fn switch_branch(&mut self, branch_name: &str) -> Result<(), CoffeeError> {
+        let repo = git2::Repository::open(&self.url.path_string)
+            .map_err(|err| error!("{}", err.message()))?;
+        let mut remote = repo
+            .find_remote("origin")
+            .map_err(|err| error!("{}", err.message()))?;
+        remote
+            .fetch(&[&branch_name], None, None)
+            .map_err(|err| error!("{}", err.message()))?;
+        let oid = repo
+            .refname_to_id(&format!("refs/remotes/origin/{}", branch_name))
+            .map_err(|err| error!("{}", err.message()))?;
+        let obj = repo
+            .find_object(oid, None)
+            .map_err(|err| error!("{}", err.message()))?;
+        repo.reset(&obj, git2::ResetType::Hard, None)
+            .map_err(|err| error!("{}", err.message()))?;
```

Comment:
Add a big `FIXME: implement the From git2 Error for `CoffeError`` this will allow to remove all the `map_err()` call
```suggestion
        // FIXME: implement the From git2 Error for `CoffeError`
        let repo = git2::Repository::open(&self.url.path_string)
            .map_err(|err| error!("{}", err.message()))?;
        let mut remote = repo
            .find_remote("origin")
            .map_err(|err| error!("{}", err.message()))?;
        remote
            .fetch(&[&branch_name], None, None)
            .map_err(|err| error!("{}", err.message()))?;
        let oid = repo
            .refname_to_id(&format!("refs/remotes/origin/{}", branch_name))
            .map_err(|err| error!("{}", err.message()))?;
        let obj = repo
            .find_object(oid, None)
            .map_err(|err| error!("{}", err.message()))?;
        repo.reset(&obj, git2::ResetType::Hard, None)
            .map_err(|err| error!("{}", err.message()))?;
```